root/usr/lib64/ruby/vendor_ruby/discovery.rb

```diff
@@ -68,6 +68,22 @@ def cmdline option=nil, default=nil
   end
 end
 
+def detect_first_nic_with_link(default = nil)
+  log_debug "Detecting the first NICs with link"
+  mac = default
```

Comment:
> I don't think this works as the last fallback because for every interface in the Dir block, you're changing the value of `mac`

Good catch, the method was implemented in a way that it returns at least
some interface if there are none with link. I am removing the default
parameter and removing the last fallback to use "macaddress" fact which
proved to be very unstable source of information. It changes over time,
reboot or even when you add new hardware via hotplug.

Therefore the link identification is the last try to get the primary
interface.

-- 
Later,
 Lukas #lzap Zapletal